web-console/src/utils/general.tsx

```diff
@@ -89,6 +89,10 @@ export function addOrUpdate<T>(xs: readonly T[], x: T, keyFn: (x: T) => string |
 
 // ----------------------------
 
+export function caseInsensitiveEquals(str1: string | undefined, str2: string | undefined): boolean {
+  return str1?.toLowerCase() === str2?.toLowerCase();
```

Comment:
This would throw a NPE if str1 or str2 are null, right? If yes, I think we should handle that as well since this is a utility function.

Reply:
no that is what the `?` before the `.` is for. It is essentially doing `(str1 ? str1.toLowerCase() : undefined) === (str2 ? str2.toLowerCase() : undefined)`. Added a test case so you can verify

Reply:
Ah I see, thanks for the clarification!